application/console.rb

```diff
@@ -70,38 +59,155 @@ def load
-        PM: proc { put_money },
-        WM: proc { withdraw_money },
-        SM: proc { send_money },
-        DA: proc { destroy_account; exit }
-    }
+    gets.chomp == YES ? create : console
   end
 
   def main_menu
     loop do
       main_menu_message
       command = gets.chomp
 
-      if commands.key?(command.to_sym)
-        commands[command.to_sym].call
-      elsif command == 'exit'
-        exit
-        break
+      case command
+      when COMMANDS[:show_cards]      then show_cards
+      when COMMANDS[:create_card]     then create_card
+      when COMMANDS[:destroy_card]    then destroy_card
+      when COMMANDS[:put_money]       then put_money
+      when COMMANDS[:withdraw_money]  then withdraw_money
+      when COMMANDS[:send_money]      then send_money
+      when COMMANDS[:destroy_account] then return destroy_account
+      when EXIT                       then break exit
       else
         puts "Wrong command. Try again!\n"
       end
     end
   end
 
+  def show_cards
+    return puts "There is no active cards!\n" unless cards
+
+    @account.current_account.card.each do |card|
+      puts "- #{card.number}, #{card.type}"
+    end
+  end
+
+  def create_card
+    loop do
+      create_card_message
+      cardtype = gets.chomp
+      card = generate_card(cardtype)
+      return puts "Wrong card type. Try again!\n" if card.nil?
+
+      cards = @account.current_account.card << card
+      @account.current_account.card = cards
+      update_accounts(@account.current_account)
+      break
+    end
+  end
+
+  def destroy_card
+    loop do
+      puts 'If you want to delete:'
+      break unless show_cards_for_operations
+
+      exit_message
+      choice = gets.chomp
+      break if choice == EXIT
+
+      return puts "You entered wrong number!\n" unless answer_validation(choice)
+
+      puts "Are you sure you want to delete #{@account.current_account.card[choice.to_i - 1].number}?[y/n]"
+      confirmation = gets.chomp
+      return unless confirmation == YES
+
+      @account.current_account.card.delete_at(choice.to_i - 1)
+      update_accounts(@account.current_account)
+      break
+    end
+  end
+
+  def put_money
+    current_card = choose_card
+    until current_card.nil?
+      amount = amount_input.to_i
+      break if amount.zero?
+
+      return high_tax unless current_card.put_tax(amount) < amount
+
+      current_card.balance = current_card.balance + amount - current_card.put_tax(amount)
+      update_accounts(@account.current_account)
+      put_money_msg(amount, current_card)
+      break
+    end
+  end
+
+  def withdraw_money
+    current_card = choose_card
+    until current_card.nil?
+      amount = amount_input.to_i
+      break if amount.zero?
+
+      return insufficient_funds if money_left(current_card, amount).negative?
+
+      update_accounts(@account.current_account)
+      withdraw_money_msg(amount, current_card)
+      break
+    end
+  end
+
+  def send_money
```

Comment:
too complex - split the logic into several methods

Reply:
i tried to make it as you said.

Reply:
tried to split and make it readable.